instrumentation/grape/lib/opentelemetry/instrumentation/grape/handler.rb

```diff
@@ -28,15 +26,68 @@ def subscribe
 
           private
 
-          def endpoint_start_process(name, start, finish, id, payload); end
+          def endpoint_run(name, start, finish, id, payload)
+            # TODO: check span.name and span.type
+            tracer.in_span(name, attributes: build_run_attributes(payload)) {}
+          end
+
+          def endpoint_render(name, start, finish, id, payload)
+            attributes = {
+              'component' => 'template',
+              'operation' => 'endpoint_render'
+            }
+            tracer.in_span(name, attributes: attributes) {}
+          end
+
+          def endpoint_run_filters(name, start, finish, id, payload)
+            filters = payload[:filters]
+            type = payload[:type]
+
+            # safe-guard to prevent submitting empty filters
+            # TODO: check if it's necessary
+            zero_length = (finish - start).zero?
+            return if (!filters || filters.empty?) || !type || zero_length
+
+            attributes = {
+              'component' => 'web',
+              'operation' => 'endpoint_run_filters',
+              'grape.filter.type' => type.to_s
+            }
+            tracer.in_span(name, attributes: attributes) {}
+          end
+
+          def build_run_attributes(payload)
+            endpoint = payload.fetch(:endpoint)
+            request_method = endpoint.options.fetch(:method).first
+            path = endpoint_expand_path(endpoint)
+            api_instance = endpoint.options[:for]
+            {
+              'component' => 'web',
+              'operation' => 'endpoint_run',
+              'grape.route.endpoint' => api_instance.base.to_s,
+              'grape.route.path' => path,
+              'grape.route.method' => request_method,
+              'http.method' => request_method,
+              'http.url' => path
+            }
```

Comment:
about these attributes we should consider using [opentelemetry::context](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/api/lib/opentelemetry/context.rb)

to store the attributes and share them across the various instrumentation points.